rlog/state-separation.mdx

```diff
@@ -0,0 +1,1255 @@
+This not only reduces costs but also increases efficiency by automating repetitive processes. The inherent security of smart contracts, combined with their 
+transparency—where every action is recorded and visible on the blockchain—makes them a powerful tool for ensuring accountability and trust in decentralized systems. 
+They can be ideal for managing decentralized autonomous organizations (DAOs), where governance decisions are automated through coded rules.
+
+By integrating smart contracts, the Nescience architecture offers a highly versatile, secure, and transparent framework that can support a wide range of applications 
+across various industries, from finance to governance, supply chains, and more.
+
+## 2. General Execution Overview
+---
+This section explains the execution process within the Nescience architecture, providing an overview of how it works from start to finish. 
+It outlines the steps involved in each execution type, guiding the reader through the entire process from user interaction to completion.
+
+The process begins when a user initiates a transaction by invoking a smart contract (SC). This invocation involves selecting at least one of 
+the four execution types: public, private, shielded, or deshielded. The choice of execution type determines how data will be read from and written to the blockchain, 
+affecting the transaction's privacy and security levels. Each execution type caters to different privacy needs, allowing the user to tailor the transaction according 
+to their specific requirements, whether it be full transparency or complete confidentiality.
+
+![general](/img/general.png)
+
+
+#### User Actions:
+
+**Step 1**: **Smart Contract Selection and Input Creation**
+- **Smart Contract Selection**: The user selects a smart contract they wish to invoke.
+- **Input Creation**: The user creates a set of inputs required for the invocation by reading the necessary data from both the public and private states. This includes:
+  - Public data such as current account balances, public keys, and smart contract states.
+  - Private data such as private account balances and UTXOs.
+
+**Step 2**: **Choosing Execution Type**
+- **Execution Type Selection**: The user selects the type of execution based on their privacy needs. The options include:
+  - **Public Execution**: Suitable for transactions where transparency is desired.
+  - **Private Execution**: Used when transaction details need to be confidential.
+  - **Shielded Execution**: Hides the receiver's identity.
+  - **Deshielded Execution**: Hides the sender's identity.
+- **zkVM Requirement**: If the execution involves private, shielded, or deshielded types, the user must call the zkVM to handle these confidential transactions. 
+For purely public executions, the zkVM is not needed, and the user can directly transmit the transaction code to the sequencer.
+
+**Step 3**: **Calling zkVM for Proof Generation**
```

Comment:
Replace:
"Calling zkVM for Proof Generation"

With:
"Calling zkVM for proof generation"